src/Native/System.Native/pal_networkchange.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_config.h"
+
+#if HAVE_LINUX_RTNETLINK_H
```

Comment:
I'm not really familiar with CMake, do we do this anywhere else? I see this block in CMakelists.txt, is this the general pattern I need to follow?

````
if (CMAKE_SYSTEM_NAME STREQUAL Linux)
    target_link_libraries(System.Native rt)
endif ()
````